File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/vocabulary/create/VocabularyCreation.js
define([ "stjs", "util/Exceptions" ], function(stjs, exceptions) {
	var VocabularyCreation = function() {
	};
	stjs.extend(VocabularyCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.vocabulary.create.VocabularyCreation';
		constructor.serialVersionUID = 1;
		prototype.code = null;
		prototype.description = null;
		prototype.managedInternally = null;
		prototype.internalNameSpace = null;
		prototype.chosenFromList = null;
		prototype.urlTemplate = null;
		prototype.terms = null;
		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description = description;
		};
		prototype.isManagedInternally = function() {
			return this.managedInternally;
		};
		prototype.setManagedInternally = function(managedInternally) {
			this.managedInternally = managedInternally;
		};
		prototype.isInternalNameSpace = function() {
			return this.internalNameSpace;
		};
		prototype.setInternalNameSpace = function(internalNameSpace) {
			this.internalNameSpace = internalNameSpace;
		};
		prototype.isChosenFromList = function() {
			return this.chosenFromList;
		};
		prototype.setChosenFromList = function(chosenFromList) {
			this.chosenFromList = chosenFromList;
		};
		prototype.getUrlTemplate = function() {
			return this.urlTemplate;
		};
		prototype.setUrlTemplate = function(urlTemplate) {
			this.urlTemplate = urlTemplate;
		};
		prototype.getTerms = function() {
			return this.terms;
		};
		prototype.setTerms = function(terms) {
			this.terms = terms;
		};
	}, {
		terms : {
			name : "List",
			arguments : [ "Object" ]
		}
	});
	return VocabularyCreation;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/vocabulary/create/VocabularyCreation.java
/*
 * Copyright 2018 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.vocabulary.create;

import java.util.List;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 *
 */
@JsonObject("as.dto.vocabulary.create.VocabularyCreation")
public class VocabularyCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    private String code;
    
    private String description;
    
    private boolean managedInternally;
    
    private boolean internalNameSpace;
    
    private boolean chosenFromList;
    
    private String urlTemplate;
    
    private List<VocabularyTermCreation> terms;

    public String getCode()
    {
        return code;
    }

    public void setCode(String code)
    {
        this.code = code;
    }

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    public boolean isManagedInternally()
    {
        return managedInternally;
    }

    public void setManagedInternally(boolean managedInternally)
    {
        this.managedInternally = managedInternally;
    }

    public boolean isInternalNameSpace()
    {
        return internalNameSpace;
    }

    public void setInternalNameSpace(boolean internalNameSpace)
    {
        this.internalNameSpace = internalNameSpace;
    }

    public boolean isChosenFromList()
    {
        return chosenFromList;
    }

    public void setChosenFromList(boolean chosenFromList)
    {
        this.chosenFromList = chosenFromList;
    }

    public String getUrlTemplate()
    {
        return urlTemplate;
    }

    public void setUrlTemplate(String urlTemplate)
    {
        this.urlTemplate = urlTemplate;
    }

    public List<VocabularyTermCreation> getTerms()
    {
        return terms;
    }

    public void setTerms(List<VocabularyTermCreation> terms)
    {
        this.terms = terms;
    }

}
